code/WorkInProgress/Cael_Aislinn/Rust/core_gen.dm

```diff
@@ -78,10 +79,26 @@ max volume of plasma storeable by the field = the total volume of a number of ti
 		assign_uid()
 		id_tag = uid
 
+/obj/machinery/power/rust_core/attackby(obj/item/weapon/W as obj, mob/living/user as mob)
+	. = ..()
+	if(.)
+		return .
+	if (istype(W, /obj/item/weapon/card/emag))
```

Comment:
There's built in code to handle engaging separately, I can't check the specifics because I'm on my phone right now though.